src/js/controller/controller.js

```diff
@@ -918,6 +918,19 @@ Object.assign(Controller.prototype, {
             return _setItem(_model.get('item'));
         };
 
+        this.setPlaylistItem = function (index, item) {
+            item = normalizePlaylistItem(_model, item, item.feedData || {});
+
+            if (item) {
+                const playlist = _model.get('playlist');
+                playlist[index] = item;
+
+                if (index === _model.get('item') && _model.get('state') === 'idle') {
+                    _setItem(index);
```

Comment:
@DanFerrer @vseventer @msiddique  Let's make sure we understand the impact on ads this has. Updating "playlistItem" on idle will kick off preloading of the item, and bids/schedules/ads. The "playlistItem" event is necessary. Making sure any previous loading is cleaned up, or exceptions are made for when the change to the item do not require new requests is follow up you will need to do for start on seek.